Models/migrations/1690328808731-databaseTeamRegistrationTriggers.ts

```diff
@@ -0,0 +1,48 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class DatabaseTeamRegistrationTriggers1690328808731 implements MigrationInterface {
+    name = "DatabaseTeamRegistrationTriggers1690328808731";
+
+    public async up (queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`
+            CREATE TRIGGER validate_qualifier_matchup
+            BEFORE INSERT ON matchup_teams_team
+            FOR EACH ROW
+            BEGIN
+                DECLARE matchupCount INT;
+                SELECT COUNT(*) INTO matchupCount
+                FROM stage
+                INNER JOIN matchup ON matchup.stageID = stage.ID
+                INNER JOIN matchup_teams_team ON matchup_teams_team.matchupID = matchup.ID
+                WHERE matchup_teams_team.teamID = NEW.teamID
+                AND stage.stageType = '0';
+                IF matchupCount > 0 THEN
+                    DELETE FROM matchup WHERE matchup.ID = NEW.matchupID;
```

Comment:
i'm 99% sure that this actually deletes the existing row instead of simply aborting the operation

Reply:
in the race condition a `matchup` row is inserted first and then a row with the keys into `matchup_teams_team`, so if there are 2 register endpoints being executed around the same time, then there ends up being 2 matchup rows and 2 matchup_teams_team rows being made

if we dont delete it then thered just end up being 2 matchup rows with 1 without any team unless im understanding it wrong id k i couldnt figure out a way to do it with `BEFORE INSERT ON matchup` instead cuz the matchup insert query itself doesnt have the team's info, so that `INSERT` query already ends up running (casuing the 2 matchups in DB), and this should only run when a team is already in a qualifier

Reply:
an improvement in this regard would be to use transactions i think? though that wouldn't completely resolve the need for a trigger (still need to enforce 1 match-up per team)

Reply:
i can look into using transactions